prefect_snowflake/database.py

```diff
@@ -97,10 +139,312 @@ def get_connection_flow():
             get_connection_flow()
             ```
         """
-        connect_params = self._get_connect_params()
+        if self._connection is not None:
+            return self._connection
+
+        connect_params = {
+            "database": self.database,
+            "warehouse": self.warehouse,
+            "schema": self.schema_,
+        }
         connection = self.credentials.get_client(**connect_kwargs, **connect_params)
         return connection
 
+    def _start_connection(self):
+        """
+        Starts Snowflake database connection.
+        """
+        self._connection = self.get_connection()
+
+    def block_initialization(self) -> None:
+        super().block_initialization()
+        if self._connection is None:
+            self._start_connection()
+
+        if self._unique_cursors is None:
+            self._unique_cursors = {}
+
+    def _get_cursor(self, inputs: Dict[str, Any]) -> Tuple[bool, SnowflakeCursor]:
+        """
+        Get a Snowflake cursor.
+
+        Args:
+            inputs: The inputs to generate a unique hash, used to decide
+                whether a new cursor should be used.
+
+        Returns:
+            Whether a cursor is new and a Snowflake cursor.
+        """
+        input_hash = hash_objects(inputs)
+        assert input_hash is not None, (
+            "We were not able to hash your inputs, "
+            "which resulted in an unexpected data return; "
+            "please open an issue with a reproducible example."
+        )
```

Comment:
Assertions can be turned off 
Assertions can be [deactivated by the user with a flag when running Python](https://docs.python.org/3/using/cmdline.html#cmdoption-O). To be sure this gets raised, we should raise a `RuntimeError`.
```suggestion
        if input_hash is None:
            raise RuntimeError(
                "We were not able to hash your inputs, "
                "which resulted in an unexpected data return; "
                "please open an issue with a reproducible example."
            )
```